post_office/models.py

```diff
@@ -138,6 +143,11 @@ def dispatch(self, log_level=None, disconnect_after_delivery=True):
         self.status = status
         self.save()
 
+        if status == STATUS.failed:
+            logger.error("email send fail: %s", self)
```

Comment:
Will those failed emails be retried automatically? If not, then I think it's at least a "warning" - as it means something is going wrong, even if it's not an "error" as it doesn't need immediate attention.